engine/api/application_key.go

```diff
@@ -10,17 +10,23 @@ import (
 	"github.com/ovh/cds/engine/api/application"
 	"github.com/ovh/cds/engine/api/event"
 	"github.com/ovh/cds/engine/api/keys"
+	"github.com/ovh/cds/engine/api/project"
 	"github.com/ovh/cds/engine/service"
 	"github.com/ovh/cds/sdk"
 )
 
 func (api *API) getKeysInApplicationHandler() service.Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
 		vars := mux.Vars(r)
-		key := vars[permProjectKey]
+		projectKey := vars[permProjectKey]
 		appName := vars["applicationName"]
 
-		app, err := application.LoadByName(api.mustDB(), key, appName, application.LoadOptions.WithKeys)
+		proj, err := project.Load(api.mustDB(), projectKey)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load project %s", projectKey)
+		}
+
+		app, err := application.LoadByProjectIDAndName(ctx, api.mustDB(), proj.ID, appName, application.LoadOptions.WithKeys)
```

Comment:
Here we just want Keys, why do we load the entire application ?
Can't we just load the keys ?

Reply:
Loading the app will first check if it exist, I will not change this for now.